test_data/module_attributes/expected_representation.txt

```diff
@@ -0,0 +1,39 @@
+defmodule Placeholder_18 do
+  @type placeholder_1 :: integer
+  @typep placeholder_2 :: integer | atom | placeholder_1 | placeholder_1
+  @opaque placeholder_3(placeholder_4) :: [{:ok, placeholder_4} | {:error, placeholder_2}]
+  @spec placeholder_5(placeholder_6, placeholder_17 :: integer) :: keyword(placeholder_6) when placeholder_6: atom
+  @spec placeholder_7(placeholder_8, placeholder_9) :: {placeholder_8, placeholder_9}
+        when placeholder_8: atom, placeholder_9: integer
+  @spec placeholder_10(placeholder_11) :: [placeholder_11] when placeholder_11: var
+  @callback placeholder_12(placeholder_19 :: placeholder_3(placeholder_1), placeholder_20 :: any) :: placeholder_10(any)
+  @macrocallback placeholder_13(placeholder_21 :: String.t()) :: Macro.t()
```

Comment:
The PR description says:

> I chose to represent all types without parentheses, as it's easy to read, the documentation uses that a lot and it made the implementation easier.

yet here's `String.t()` and `Macro.t()`. IMO it doesn't matter, but I just wanted to point out in case it's a bug in your opinion.

Reply:
Whoops, I missed that. Yeah, I would consider that a bug (or at the very least, it would bug me)

Reply:
Well... Looks like I'm going to have to live with it. I do have code that sets `no_parens: true` in the metadata at the right place, but it looks like this is something the formatter feels strong enough to overwrite (in `Macro.to_string` since 1.13), and I have not found a way to change it.